torch/testing/_internal/common_methods_invocations.py

```diff
@@ -349,6 +349,80 @@ def wrapped_fn(x):
 
     return wrapped_fn
 
+def np_binary_ufunc_type_promotion_wrapper(fn):
```

Comment:
I appreciate the attention to detail with this function, and we may bring it back later, but for this PR a simpler approach is probably to remove this function and the following two and, instead, just use torch.result_type and pre-cast all the NumPy inputs to the corresponding NumPy dtype. For example:

```
dtype = torch.result_type(input0, input1)
np_input0 = input0.to(dtype).cpu().numpy()
np_input1 = input1.to(dtype).cpu().numpy().
```

This is imperfect since the result type doesn't actually impact the computation like pre-casting will, but I believe this will be a practical solution for this PR. Let me know if you have questions about this; we can walk through it together.

Reply:
I will choose the second solution because it's still required some hard code for the corner cases for arguments with tensor and python number.